Optimize getProxyMode() to get proxy mode

FengyunPan opened the pull request:
1. getProxyMode() does not need nodeGetter args after remove
   proxy-mode annotation.
2. we should get error when the version of iptables less than
   MinVersion.

k8s-reviewable commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.kubernetes.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/44204)
<!-- Reviewable:end -->

FengyunPan commented:
/release-note-none

thockin commented:
/approve

thockin commented:
@freehan for review

justinsb commented on pkg/proxy/iptables/proxier.go:
```diff
@@ -110,7 +110,7 @@ func CanUseIPTablesProxier(iptver IPTablesVersioner, kcompat KernelCompatTester)
 	}
 	if version.LessThan(minVersion) {
-		return false, nil
+		return false, fmt.Errorf("The iptables version at least %s", iptablesMinVersion)
```
"The iptables version must be at least %s"?

And does this belong in this PR?

FengyunPan replied:
Thank you. It makes sense, I will change it soon.

freehan commented:
@k8s-bot ok to test

Commit:
Optimize getProxyMode() to get proxy mode

The getProxyMode() does not need nodeGetter args after remove
proxy-mode annotation.

freehan commented:
Thanks!
/lgtm

freehan commented:
@k8s-bot gce etcd3 e2e test this
